react/index.js

```diff
@@ -110,76 +118,40 @@ class CategoryMenu extends Component {
   renderMenu() {
     const {
       data: { categories = [] },
-      intl,
       showAllDepartments,
       showSubcategories,
       menuPosition,
-      runtime,
+      additionalItems,
+      intl,
     } = this.props
 
-    const pathName = path(['route', 'params', 'department'], runtime)
-
-    const department = pathName ? pathName : ''
-
-    const desktopClasses = classNames(
-      `${categoryMenu.container} w-100 bg-base dn flex-m`,
-      {
-        'justify-start mw9':
-          menuPosition === categoryMenuPosition.DISPLAY_LEFT.value,
-        'justify-end mw9':
-          menuPosition === categoryMenuPosition.DISPLAY_RIGHT.value,
-        'justify-center':
-          menuPosition === categoryMenuPosition.DISPLAY_CENTER.value,
-      }
-    )
-
     return (
-      <Container className="justify-center flex">
-        <nav className={desktopClasses}>
-          <ul className="pa0 list ma0 flex flex-wrap flex-row t-action overflow-hidden h3">
-            {showAllDepartments && (
-              <CategoryItem
-                noRedirect
-                menuPosition={menuPosition}
-                subcategoryLevels={
-                  DEFAULT_SUBCATEGORIES_LEVELS + showSubcategories
-                }
-                category={{
-                  children: categories,
-                  name: intl.formatMessage({
-                    id: 'category-menu.departments.title',
-                  }),
-                }}
-              />
-            )}
-            {this.departments.map(category => (
-              <Fragment key={category.id}>
-                <CategoryItem
-                  menuPosition={menuPosition}
-                  category={category}
-                  subcategoryLevels={
-                    DEFAULT_SUBCATEGORIES_LEVELS + showSubcategories
-                  }
-                  isCategorySelected={department === category.slug}
-                />
-              </Fragment>
-            ))}
-          </ul>
-        </nav>
-      </Container>
+      <Menu
+        categories={categories}
+        showAllDepartments={showAllDepartments}
+        menuPosition={menuPosition}
+        subcategoryLevels={DEFAULT_SUBCATEGORIES_LEVELS + showSubcategories}
+        departments={this.departments}
+        departmentsTitle={intl.formatMessage({
+          id: 'category-menu.departments.title',
+        })}
+        additionalItems={additionalItems}
+      />
     )
   }
 
   render() {
-    const { mobileMode } = this.props
+    const {
+      runtime: {
+        hints: { mobile },
+      },
+    } = this.props
 
-    return mobileMode ? this.renderSideBar() : this.renderMenu()
+    return mobile ? this.renderSideBar() : this.renderMenu()
```

Comment:
⛏ I would inline those functions, like: `mobile ? <Sidebar .../> : <Menu .../>`. Specially now that both are easy to read.